common/lib/dependabot/package_manager.rb

```diff
@@ -0,0 +1,94 @@
+# typed: strict
+# frozen_string_literal: true
+
+require "sorbet-runtime"
+
+module Dependabot
+  class PackageManagerBase
+    extend T::Sig
+    extend T::Helpers
+
+    abstract!
+
+    # The name of the package manager (e.g., "bundler").
+    # @example
+    #   package_manager.name #=> "bundler"
+    sig { abstract.returns(String) }
+    def name; end
+
+    # The version of the package manager (e.g., "2.1.4").
+    # Returns nil if the version is not set.
```

Comment:
Wondering if we can make this stricter? To always return Versions instead of nilable Versions